Added pyray stub

Guilherme32 opened the pull request:
Adding a pyray stub so that IDEs can use autocompletion and not show unresolved references on the code

electronstudio commented:
How did you generate this? I’m reluctant to put anything in that can’t be auto generated because I won’t have time to update it manually on each new release.

Guilherme32 commented:
So... I kinda did it mannualy. I don't know how it could be done on build time, but this is how I did it. I first imported this handy function from the inspect library:

``` python
from inspect import signature
```

Then I added some prints to the part of the raypy.py script that adds the attributes, in order to build the text for the stub:

``` python
for name, attr in getmembers(rl):
    # print(name, attr)
    uname = inflection.underscore(name).replace('3_d', '_3d').replace('2_d', '_2d')
    if isbuiltin(attr) or str(type(attr)) == "<class '_cffi_backend.__FFIFunctionWrapper'>":
        # print(attr.__call__)
        # print(attr.__doc__)
        # print(attr.__text_signature__)
        # print(dir(attr))
        # print(dir(attr.__repr__))
        f = makefunc(attr)
        setattr(target, uname, f)

        sig = signature(f)
        return_type = str(attr).split(' ')[1][1:]           # this isolates the type
        
        if return_type == 'unsigned':
         return_type = str(attr).split(' ')[2]
        
        if return_type == 'void':
            return_type = 'None'
            
        print(f"def {uname}{sig} -> {return_type}: ...")

        # def wrap(*args):
        #    print("call to ",attr)
        # setattr(PyRay, uname, lambda *args: print("call to ",attr))
    else:
        print(f"{name}: {str(type(attr))[8:-2]}")           # this isolates the type
        setattr(target, name, attr)

for struct in ('Vector2', 'Vector3', 'Vector4', 'Camera2D', 'Camera3D', 'Quaternion', 'Color', 'Rectangle'):
    f = makeStructHelper(struct)
    print(f"{struct}: struct")
    setattr(target, struct, f)
```

After importing raypy, I copied the entire block from my command line and pasted onto the newly created stub, under the RayPy class, using mass indent to put it all in the proper scope. I then commented these lines for it to not print again. I imagine it wouldn't be so hard to save these lines to a file instead of printing, but I don't know how to do it automatically on github...
